Moible Api Endpoint

Mahamudul42 opened the pull request.

kluver commented on mobile_api.py:
```diff
@@ -0,0 +1,170 @@
+
+@mobile_api.route("/newsletters", methods=["GET"])
+@auth.requires_login
+def get_newsletters():
```
Have not reviewed

karlhigley commented on mobile_api.py:
```diff
@@ -0,0 +1,170 @@
+                    "articles": articles,
+                }
+            )
+        conn.commit()
```
Since this function is only fetching data and not storing anything, I don't think it needs to commit a transaction:
```suggestion
```

Mahamudul42 replied:
Okay, I will update that.

karlhigley reviewed (commented):
This looks like a viable way to put together an API for a demo. Left a few comments about code clean-up, which are nice-to-have but not critical for short-lived code.

The one concern that would stop from me merging this (aside from any auth/tokens concerns @kluver raised) is around account creation without the consent and onboarding process. I think the lowest effort way address that concern is to adapt the code to look for existing accounts and return an error if an account with a matching email address isn't found.

karlhigley commented on mobile_api.py:
```diff
@@ -0,0 +1,170 @@
+        newsletter_repo = DbNewsletterRepository(conn)
+        image_repo = DbImageRepository(conn)
+        accounts = [Account(account_id=account_id)]
+        newsletters = newsletter_repo.fetch_newsletters_since(days_ago=30, accounts=accounts)
+        # Filter to get only the latest newsletter
+        if newsletters:
+            newsletters = [max(newsletters, key=lambda n: n.created_at)]
+        result = []
+        for newsletter in newsletters:
```
I don't think we're actually allowed to do that because IIRC we're not supposed to display AP articles after the first 14 days beyond their publication date. @jakonstan could confirm or deny that though, and a more limited time window would presumably still work.

Mahamudul42 commented:
This PR addresses feedback. Key changes include:

1. Moved get_account_by_email to postgres_db.py for better modularity.
2. Added get_account_by_email method to the Auth class in auth.py to ensure proper access.
3. Updated mobile_api.py to use days_ago=1 for fetching latest newsletter.

karlhigley commented on util/auth.py:
```diff
@@ -208,5 +207,9 @@ def send_post_consent(self):
+    def get_account_by_email(self, email):
+        # Retrieve an account by email address using the postgres_db module.
+        return get_account_by_email(email)
+
```
@kluver Does this way of fetching accounts by email address make sense to you? I'm a little puzzled because I don't see that putting a method here does anything to restrict access and I'm not sure if we need a separate utility for fetching accounts by email address or if an existing repository method would work here. That might just be because I'm less familiar with the auth code though

kluver commented on util/postgres_db.py:
```diff
@@ -60,6 +60,19 @@ def get_account(account_id):
             return result
 
 
+def get_account_by_email(email):
```
If we need to have this, we should make sure it returns the same things as getAccounts above it. (teams and experiments)

kluver commented:
I think this is moving towards good, but after reading Karl's feedback I think I agree we would want to check that your endpoint for setting up for mobile use doesn't have any code that could create new users and also directly checks if users have finished onboarding.
